src/Microsoft.ML.LightGBM/LightGbmArguments.cs

```diff
@@ -321,6 +321,11 @@ public enum EvalMetricType
         [TGUI(Label = "Ranking Label Gain")]
         public string CustomGains = "0,3,7,15,31,63,127,255,511,1023,2047,4095";
 
+        [Argument(ArgumentType.AtMostOnce, HelpText = "Parameter for the sigmoid function. Used only in binary and multiclassova " +
+            "classification and in lambdarank applications", ShortName = "sigmoid")]
```

Comment:
>lambdarank [](start = 35, length = 10)

just ranking.

Reply:
lambda rank is the ranking objective function.

Reply:
Lambda rank and multiclassova is internal Lightgbm terminology. Consider user who never heard about lightgbm, but want to give it a try. For this user we don't expose anything related to lambda ranking or multiclassova, we have our own terminology of ranking trainer, or multiclassification, and we should use it.

---
In reply to: [208723780](https://github.com/dotnet/machinelearning/pull/667#discussion_r208723780) [](ancestors = 208723780)

Reply:
LamdaRank is a famous ranking algorithm, it is not just used in LightGBM but also in fasttree. Read more here: https://www.microsoft.com/en-us/research/publication/from-ranknet-to-lambdarank-to-lambdamart-an-overview/

Reply:
I have no doubt what lambdarank is famous algorithm, it's more a question of how often we use it in any user facing api?
We do have multiclass classification, binary classification, ranking, we have LightGbmRanker class, but how often user see word lambdarank? 
Look from user perspective, are they same, or there is difference.

All I'm trying to say, I don't understand what's the point of copying documentation from lightgbm site which is confusing in ML.Net (this is two different projects) if we can just rephase it in more ML.net appropriate terminology.

---
In reply to: [208727880](https://github.com/dotnet/machinelearning/pull/667#discussion_r208727880) [](ancestors = 208727880)

Reply:
Your concern as I understand from your second comment on this thread has to do with "Lambda rank and multiclassova is internal Lightgbm terminology" and as I have pointed out previously this statement is not accurate. 

Now lets talk about your other concerns that you have brought up in your most recent comment, I think we should change the documentation where there is a need to and we should also try to keep the parameter list and documentation as close to LightGBM's official documentation so that our users that have been using command line and python implementation of lightGBM are not confused and its easy for them. Given there are [so many](https://en.wikipedia.org/wiki/Learning_to_rank) ranking algorithms out there it is better to be specific about which one we are exposing than having user ask questions or figure themselves out by reading code or however.

Reply:
pipeline.Add(new LightGbmRanker(...));
Can I use sigmoid in arguments, or it's useless?
Is it lambdarank or not? And more specifically, how user find answer on this question?

---
In reply to: [208734818](https://github.com/dotnet/machinelearning/pull/667#discussion_r208734818) [](ancestors = 208734818)

Reply:
Reactivating. For intellisense, it would be better if the argument isn't relevant to that trainer that it simply not appear *at all*.

---
In reply to: [208740753](https://github.com/dotnet/machinelearning/pull/667#discussion_r208740753) [](ancestors = 208740753)